packages/roosterjs-editor-types/lib/interface/EditorCore.ts

```diff
@@ -264,6 +277,13 @@ export interface CoreApiMap {
      */
     getStyleBasedFormatState: GetStyleBasedFormatState;
 
+    /**
+     * Get style based format state from current selection, including font name/size and colors
+     * @param core The EditorCore objects
+     * @param node The node to get style from
+     */
+    getPendingFormatState: GetPendingFormatState;
```

Comment:
Should we remove the existing getPendableFormatState? I just tested getPendingFormatState for the execCommand scenario. The getPendableFormatState function is used in createPasteFragment and select Range and I didn't test getPendingFormarState in this scenarios.

Reply:
You bring up a very good question.

createPasteFragment calls getPendableFormatState() to determine what format to merge. So it should always read format from DOM tree, even if there is cached pending format. So that we need to add a parameter to the new API to allow force getting stats from DOM tree, then let createPasteFragment use it.

After that, the original getPengingFormatState function will not be used any more, so we can mark it as deprecated, but keep it for now. We will remove it in next major release.